pages/metadataExport/MetadataExportHandler.inc.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * @file pages/metadataExport/MetadataExportHandler.inc.php
+ *
+ * Copyright (c) 2013-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class MetadataExportHandler
+ * @ingroup pages_metadataExport
+ *
+ * @brief Handle MetadataExportHandler requests.
+ */
+
+import('pages.manager.ManagerHandler');
+
+class MetadataExportHandler extends ManagerHandler {
+	
+	/* @var journal Journal object - the current journal */
+	var $journal;
+	
+	/* @var articles Array - the articles of the exported issue/journal */
+	var $articles;
+	
+	/* @var fileName String - the name of the export file */
+	var $fileName;
+	
+	/* @var publishedArticleDao PublishedArticleDAO object */
+	var $publishedArticleDao;
+	
+	/* @var issueDao IssueDAO object */
+	var $issueDao;
+	
+	/* @var metadataExportPlugins Array - all plugins of the metadataexport category */
+	var $metadataExportPlugins;
+	
+	/* @var exportFunction String - the function defining the scope of the export (article/issue/journal) */
+	var $exportFunction;
+	
+	/* @var exportPluginName String - the name of the current export plugin */
+	var $exportPluginName;
+	
+	/* @var exportPlugin MetadataExportPlugin object - the current export plugin object */
+	var $exportPlugin;
+	
+	/* @var elementId Int - the id of the current article/issue */
+	var $elementId;
+	
+	
+	/**
+	 * @function MetadataExportHandler - Constructor
+	 * @param $request Request
+	 */
+	function MetadataExportHandler($request) {
+		parent::ManagerHandler($request);
+		
+		$this->exportFunction = '_' . $request->getUserVar('exportScope');
```

Comment:
By convention, we're using "_" as a prefix for private functions, and here it's used to expose functions for arbitrary execution. I'd rather we didn't fetch function names from request variables -- there's a lot of security risk for not much benefit.